deepchecks/tabular/checks/integrity/outlier_detection.py

```diff
@@ -0,0 +1,149 @@
+        else:
+            dataset = context.test
+        dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
+        df = select_from_dataframe(dataset.data, self.columns, self.ignore_columns)
+
+        # Calculate distances matrix and retrieve nearest neighbors based on distance matrix.
+        cat_features = df.columns.map(lambda x: x in dataset.cat_features)
+        dist_matrix = gower_distance.gower_matrix(np.asarray(df), cat_features=np.array(cat_features))
+        knn_distance_based = NearestNeighbors(n_neighbors=self.num_nearest_neighbors, metric='precomputed').fit(
+            dist_matrix)
+        nn_matrix = knn_distance_based.kneighbors(return_distance=False)
+
+        # Calculate outlier probability score using loop algorithm.
+        m = loop.LocalOutlierProbability(distance_matrix=dist_matrix, neighbor_matrix=nn_matrix,
+                                         extent=self.extend_parameter, n_neighbors=self.num_nearest_neighbors).fit()
+        prob_vector = m.local_outlier_probabilities
+
+        # Create the check result visualization
+        top_n_idx = np.argsort(prob_vector)[-self.n_to_show:]
+        top_n_values = [prob_vector[i] for i in top_n_idx]
+        dataset_outliers = df.iloc[top_n_idx, :]
+        dataset_outliers.insert(0, 'Outlier Probability Score', top_n_values)
+        dataset_outliers.sort_values('Outlier Probability Score', ascending=False, inplace=True)
+
+        headnote = """<span>
+                    The Outlier Probability Score is calculated by the LoOP algorithm which measures the local deviation
+                     of density of a given sample with respect to its neighbors. These outlier scores are directly
+                      interpretable as a probability of an object being an outlier. see 
+                      <a href="https://www.dbs.ifi.lmu.de/Publikationen/Papers/LoOP1649.pdf" target="_blank" rel="noopener noreferrer">link</a>
+                       for more information.<br><br>
+                </span>"""
+
+        quantiles_vector = np.quantile(prob_vector, np.array(range(1000)) / 1000, method='closest_observation')
+        return CheckResult([quantiles_vector, df.shape[0]], display=[headnote, dataset_outliers])
+
+    def add_condition_not_more_outliers_than(self, max_outliers_ratio: float = 0.005,
+                                             outlier_score_threshold: float = 0.8):
+        """Add condition - no more than given ratio of elements over outlier threshold are allowed.
+        Parameters
+        ----------
+        max_outliers_ratio : float , default: 0.005
+            Maximum ratio of outliers allowed in dataset.
+        outlier_score_threshold : float, default: 0.8
+            Outlier probability score threshold to be considered outlier.
+        """
+        if max_outliers_ratio > 1 or max_outliers_ratio < 0:
+            raise ValueError('max_outliers_ratio must be between 0 and 1')
```

Comment:
```suggestion
            raise DeepchecksValueError('max_outliers_ratio must be between 0 and 1')
```
This makes sure it won't kill the suite it's running inside.